Automattic/wp-calypso
# Copy Site: Add back functionality on domains step

See p4TIVU-aoH-p2#comment-11701
Related to https://github.com/Automattic/dotcom-forge/issues/1519

Original pr: https://github.com/Automattic/wp-calypso/pull/72327

#### Proposed Changes

Refactor of the `DomainsStep` component to solve the bug where the user needs to click twice to the bug button to go back to the `/sites` page.

The step has been split into two components: the main component which renders the step wrapper (which includes the header), and a child component which renders the domain picker. The component that renders the header no longer needs to subscribe to the `domainForm` value from the onboarding store. Each time the domain search field is unfocused `domainForm` gets resaved, and this is what was causing the header to rerender.

#### Testing Instructions


##### Test Copy site flow
1. Make sure you have a site with `Business` plan and atomic.
2. Navigate to `/sites` 
3. Click the dots menu on the right (at your site's row)
4. You should be navigated to the domain selection screen.
5. On the top right there is a `Back` button. Click it **once**, and ensure that you navigate back to the `/sites` page

##### Test Link in bio flow

Because of the refactor, makes sense to take a look at `link-in-bio` flow as well to see that the component works as expected:

1. Navigate to `/setup?flow=link-in-bio`
2. Take the flow until you land to the domain selection screen.
3. Ensure that the input is working (search for a domain and ensure that it works)
4. Test that the `Decide later` button works (should skip this step)
5. Make sure that the filters are working.
6. Select a free domain and ensure that the process completes successfully.
7. Ensure you select a free domain and a plan other than Free. Make sure you can checkout successfully.

#### Pre-merge Checklist





fixes https://github.com/Automattic/dotcom-forge/issues/1519
